### PR TITLE
refactor: provision-react-component

### PR DESCRIPTION
**Pull request by keithamus**
https://keithamus.github.io/component-header

### CODE REVIEW

**Comment by keithamus** [comment]
@danielmoll updated to new PRC.

File: src/blogCustomisation.css [hunk 1]
```diff
@@ -10,39 +8,40 @@
...
+    calc(100vw / var(--header-mobile-image-aspect-ratio, 0.55));
 }
 
+/* stylelint-enable selector-class-pattern */
+
 @media (--header-viewport-big) {
-  .header__follow-us-title,
-  .header__follow-us-icons {
-    line-height: 40px;
+  .header__follow-us-title {
```
Comment (danielmoll) [inline src/blogCustomisation.css]: Any reason why this block is not like below given they share a majority of styles? It's not exactly important just curious :)
```
.header__follow-us-title,
.header__follow-us-icons {
  display: inline-block;
  vertical-align: middle;
  line-height: 40px;
}
.header__follow-us-icons {
  padding: 0 var(--grid-gutter-s);
}
```

Comment (keithamus) [inline src/blogCustomisation.css]: Our stylelint rules stop us from duplicating selectors (in favour of instead duplicating properties). Duplicating selectors makes it hard to track what properties are applied to what selector. Duplicate properties on the other hand have no cognitive overload and also gzip well :smile:

File: test/index.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      shallowRenderer.render(React.createElement(Header, {
+        title: 'Required',
+        flyTitle: 'Flytitle',
+        text: 'Text',
+      }));
+      header = shallowRenderer.getRenderOutput();
+      const text = header.props.children.props.children.props.children.props.children[2];
+      text.props.className.should.equal('header__text');
+    });
+  });
```
Comment (danielmoll) [inline test/index.js]: Could add tests to cover off the large/small class `'header--small-mode': Boolean(smallMode) === true` fix?

Something like:
```
it('it renders with the className header--large-mode by default', () => {
  shallowRenderer.render(React.createElement(Header, {
    title: 'Required',
    flyTitle: 'Flytitle',
    text: 'Text',
  }));
  header = shallowRenderer.getRenderOutput();
  header.props.className.should.equal('header header--large-mode');
});
it('it renders with the className header--small-mode if smallMode is true', () => {
  shallowRenderer.render(React.createElement(Header, {
    title: 'Required',
    flyTitle: 'Flytitle',
    text: 'Text',
    smallMode: true,
  }));
  header = shallowRenderer.getRenderOutput();
  header.props.className.should.equal('header header--small-mode');
});
```

**Comment by danielmoll** [comment]
Otherwise all good, happy to merge :)

**Pull request merged**